### PR TITLE
[UH-402] WIP, Add trigger and validation

### PR DESCRIPTION
**Pull request by pabloqc**

### CODE REVIEW

File: lib/baustelle/config/validator/application.rb [hunk 1]
```diff
@@ -44,6 +44,11 @@ def schema
               'min' => Fixnum,
               'max' => Fixnum
             },
+            optional('trigger') => {
+              optional('MeasureName') => String,
```
Comment (sevos) [inline lib/baustelle/config/validator/application.rb]: I would not make these 2nd level fields optional, but only the parent "trigger". By that you will enforce that if trigger is defined - all three keys are required

File: lib/baustelle/config/validator/application.rb [hunk 2]
```diff
@@ -44,6 +44,12 @@ def schema
               'min' => Fixnum,
               'max' => Fixnum
             },
+            # https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/command-options-general.html#command-options-general-autoscalingtrigger
+            optional('trigger') => {
+              'MeasureName' => String,
+              'LowerThreshold' => Fixnum,
+              'UpperThreshold' => Fixnum
```
Comment (suxor42) [inline lib/baustelle/config/validator/application.rb]: Please stick snake case for the YAML

File: lib/baustelle/config/validator/application.rb [hunk 3]
```diff
@@ -44,6 +44,12 @@ def schema
               'min' => Fixnum,
               'max' => Fixnum
             },
+            # https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/command-options-general.html#command-options-general-autoscalingtrigger
+            optional('trigger') => {
+              'MeasureName' => String,
```
Comment (suxor42) [inline lib/baustelle/config/validator/application.rb]: Check for correct names and only allow the ones that are possible

**Comment by suxor42** [comment]
Tests are missing

File: lib/baustelle/config/validator/application.rb [hunk 4]
```diff
@@ -28,6 +29,7 @@ def schema_walk(value, mapper)
         end
 
         include RSchema::DSL::Base
+        include RSchema::AWSAutoscalingTriggers
```
Comment (sevos) [inline lib/baustelle/config/validator/application.rb]: I would not include this module here. Unnecessary indirection.

Comment (sevos) [inline lib/baustelle/config/validator/application.rb]: reference `RSchema::AWSAutoscalingTriggers::MEASURES` directly

**Comment by sevos** [comment]
reviewed

**Comment by suxor42** [comment]
I just added a testcase. Looks fine

**Pull request merged**